bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/EventFilterHandler.java

```diff
@@ -257,6 +254,9 @@ private void updateChannelSet(EventFilterConfiguration config) {
      * Updates all states and channels. Reschedules an update if no error occurs.
      */
     private void updateStates() {
+        if (!initFinished) {
+            logger.debug("Ignoring call for updating states as this instance is not initialized yet.");
+        }
```

Comment:
Shouldn't you return here?

Reply:
Yep. This happens if changing sources at multiple locations at a time...